### ISSUE TITLE
Numpy Gamma distribution not working in Numba 0.57+

### ISSUE THREAD

**Issue by devindg** [numba/numba#9053]
## Reporting a bug

- [ x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/main/CHANGE_LOG).
- [ x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

Vectorization of Numpy's gamma distribution no longer works. The below code works in Numba 0.56.4, but not in 0.57.0 or 0.57.1. 

```
from numba import float64, vectorize
from numpy.random import gamma
import numpy as np

@vectorize([float64(float64, float64)])
def vec_ig(shape, scale):
    ig = 1. / gamma(shape=shape, scale=1. / scale)
    return ig

a = np.array([0.1, 0.1], dtype=np.float64)
b = np.array([0.1, 0.1], dtype=np.float64)
res = vec_ig(a, b)
```

The error that I receive is:

```
Compilation is falling back to object mode WITHOUT looplifting enabled because Function "vec_ig" failed type inference due to: No implementation of function Function(<built-in method gamma of numpy.random.mtrand.RandomState object at 0x7f7ae5b65840>) found for signature:
 
 >>> gamma(shape=float64, scale=float64)
 
There are 6 candidate implementations:
  - Of which 6 did not match due to:
  Overload in function 'gammavariate_impl': File: numba/cpython/randomimpl.py: Line 774.
    With argument(s): '(shape=float64, scale=float64)':
   Rejected as the implementation raised a specific error:
     TypingError: missing a required argument: 'alpha'
  raised from /home/devin/.cache/pypoetry/virtualenvs/pybuc-opbxzPBX-py3.9/lib64/python3.9/site-packages/numba/core/typing/templates.py:784
```

Expected behavior:

Return an array of size 2 with random draws from the gamma distribution.

**Comment by esc** [comment]
@devindg thank you for this report, I can confirm that this works on 0.56.4 and doesn't on current `HEAD` at 249c8ff3206928b486346443ec148508f8c25f8e

**Comment by esc** [comment]
I think this is a regression caused by: https://github.com/numba/numba/issues/8061

**Comment by apmasell** [comment]
This is not exactly a regression. NumPy and CPython have some of the same distributions and we share code, but they have different names for the parameters (`shape` vs `alpha` and `scale` vs `beta` in the case of the gamma distribution). This means if you use named parameters, everything falls apart and if you use positional parameters, everything is fine. Since overloads needs matching parameter names all the way down, this means that switching to overloads merged names. All our tests use only positional parameters, so they behave as expected.

The quick fix is to use positional parameters. I'm not sure what the best long term approach is.

**Comment by esc** [comment]
I think there is now a fix at: https://github.com/numba/numba/pull/9059

**Comment by gmarkall** [comment]
I think this was fixed by #9111, which superseded #9099, which superseded #9059. The reproducer runs to completion with `main` (and I have verified that it also did not in my setup with 0.57.1).

**Comment by esc** [comment]
@gmarkall thank you for checking on this and following up.
